apisix/init.lua

```diff
@@ -263,6 +263,16 @@ local function set_upstream_headers(api_ctx, picked_server)
     if x_forwarded_host then
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
+
+    local x_forwarded_for = api_ctx.var.http_x_forwarded_for
+    if x_forwarded_for then
+        api_ctx.var.http_x_forwarded_for = x_forwarded_for .. ", " .. api_ctx.var.realip_remote_addr
```

Comment:
One can remove the var value in serverless plugins, so please check the value before concating it.

Reply:
Done